processor-steps/src/main/java/boringyuri/processor/common/steps/IndependentUriDataGeneratorStep.kt

```diff
@@ -195,37 +196,49 @@ class IndependentUriDataGeneratorStep(
         }
 
         if (method.parameters.isNotEmpty()) {
-            val parameters = method.parameters.joinToString { it.simpleName }
+            val parameters = method.parameters.joinToString { it.name }
             logger.warn(method, "Method parameters [$parameters] will be ignored")
         }
 
         return true
     }
 
     private fun generateUriDataClass(
-        sourceElement: TypeElement,
+        sourceElement: XTypeElement,
         uriMetadata: UriMetadata
     ): Boolean {
-        val packageElement = elementUtils.getPackageOf(sourceElement)
-        val packageName = packageElement.qualifiedName.toString()
-        val simpleClassName = sourceElement.simpleName.toString() + CONTAINER_IMPL_SUFFIX
+        val packageName = sourceElement.packageName
+        val simpleClassName = sourceElement.name + CONTAINER_IMPL_SUFFIX
         val className = ClassName.get(packageName, simpleClassName)
 
         val content = generateUriDataClassContent(
             className,
             sourceElement,
             uriMetadata,
-            TypeName.get(sourceElement.asType())
+            sourceElement.type.typeName
         )
 
-        writeSourceFile(className, content, sourceElement)
+        session.fileWriter.writeSourceFile(className, content, XFiler.Mode.Isolating)
 
         return true
     }
 
-    private companion object {
-        const val CONTAINER_IMPL_SUFFIX = "Impl"
-        val GETTER_PATTERN = "^(?:get|is|has|are)([A-Z][a-zA-Z0-9]+)$".toRegex()
+    companion object {
+        private const val CONTAINER_IMPL_SUFFIX = "Impl"
+        private val GETTER_PATTERN = "^(?:get|is|has|are)([A-Z][a-zA-Z0-9]+)$".toRegex()
+
+        private val INTERNAL_ANNOTATIONS: Set<TypeName> = hashSetOf(
+            CommonTypeName.OVERRIDE,
+            ClassName.get(UriData::class.java),
+            ClassName.get(Path::class.java),
+            ClassName.get(Param::class.java),
+            ClassName.get(DefaultValue::class.java),
+            ClassName.get(TypeAdapter::class.java)
+        )
```

Comment:
nit: to make more sense to your `CommonInternalAnnotations.kt` you may move this constant there as well under a name `INDEPENDENT_DATA_ANNOTATIONS` or something like this.

Reply:
done